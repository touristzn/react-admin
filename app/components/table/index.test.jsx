import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Table, TableColumn } from './index'

// 使用方法
//-----------------------------
// 自适应效果
// 在Tabel上加width="100%"
// 不传宽则宽度等于所有列宽度的和
//-----------------------------
// 横向滚动条效果
// Tabel上如果不传width，子组件TableColumn则需要传固定width
// 如果想要某列自适应宽度，可以不传宽，或为空

let wrapper

describe('<Table />', () => {
  const props = {
    data: [
      {
        "id": 1,
        "title": "去云南旅游1",
        "price": 200,
        "date": "2018-08-01",
      },
      {
        "id": 2,
        "title": "去云南旅游2",
        "price": 300,
        "date": "2018-08-01",
      },
    ],
    width: '100%',
    className: 'custom',
  };

  beforeEach(() => {
    wrapper = shallow(
      <Table {...props}>
        <TableColumn prop="title" label="内容" />
        <TableColumn prop="price" width="100" label="日期" />
        <TableColumn
        label="创建时间"
        width="150"
        render={(data) => {
          return <span>{ data.date }</span>
        }}
        />
        <TableColumn>
          <span>修改</span>
          <span>查看</span>
        </TableColumn>
      </Table>
    )
  })

  it('should render the index', () => {
    const state = {
      page: 1,
      pageSize: 20,
    }

    const indexMethod = (index) => {
      const { page, pageSize } = state
      return index + (page - 1) * pageSize + 1;
    }

    const tree = shallow(
      <Table {...props}>
        <TableColumn index={indexMethod} width="30" label='序号' />
      </Table>
    )

    expect(tree.find('tbody td').at(1).text()).toBe('2')
  })

  it('should contain a custom class', () => {
    expect(wrapper.hasClass('custom')).toBeTruthy()
  })

  it('should render the correct items', () => {
    expect(wrapper.find('thead th')).toHaveLength(4)
    expect(wrapper.find('tbody tr')).toHaveLength(props.data.length)
    expect(wrapper.find('thead th').at(0).text()).toBe('内容')
    expect(Number(wrapper.find('tbody td').at(1).text())).toBe(props.data[0].price)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})