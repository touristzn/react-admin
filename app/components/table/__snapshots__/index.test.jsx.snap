// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
<div
  className="table-list custom"
>
  <table
    border="0"
    cellPadding="0"
    cellSpacing="0"
    width="100%"
  >
    <colgroup>
      <col
        key=".0"
      />
      <col
        key=".1"
        width="100"
      />
      <col
        key=".2"
        width="150"
      />
      <col
        key=".3"
      />
    </colgroup>
    <thead>
      <tr>
        <th
          key=".0"
        >
          内容
        </th>
        <th
          key=".1"
        >
          日期
        </th>
        <th
          key=".2"
        >
          创建时间
        </th>
        <th
          key=".3"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <td
          key=".0"
        >
          去云南旅游1
        </td>
        <td
          key=".1"
        >
          200
        </td>
        <td
          key=".2"
        >
          <span>
            2018-08-01
          </span>
        </td>
        <td
          key=".3"
        >
          <TableColumn>
            <span>
              修改
            </span>
            <span>
              查看
            </span>
          </TableColumn>
        </td>
      </tr>
      <tr
        key="1"
      >
        <td
          key=".0"
        >
          去云南旅游2
        </td>
        <td
          key=".1"
        >
          300
        </td>
        <td
          key=".2"
        >
          <span>
            2018-08-01
          </span>
        </td>
        <td
          key=".3"
        >
          <TableColumn>
            <span>
              修改
            </span>
            <span>
              查看
            </span>
          </TableColumn>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
